src/app/routes/applications/new-role/new-role.component.spec.ts

```diff
@@ -1,25 +1,259 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
 
-import { NewRoleComponent } from './new-role.component';
+import { provideMockStore } from '@ngrx/store/testing';
 
-xdescribe('NewRoleComponent', () => {
+import { NewRoleComponent, RoleType } from './new-role.component';
+import { IamService } from '../../../shared/services/iam.service';
+import { ToastrService } from 'ngx-toastr';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { PreconditionTypes } from 'iam-client-lib';
+
+describe('NewRoleComponent', () => {
   let component: NewRoleComponent;
   let fixture: ComponentFixture<NewRoleComponent>;
+  const fb = new FormBuilder();
+  const matDialogSpy = jasmine.createSpyObj('MatDialog',
+      [
+        'closeAll',
+      ]);
+  const toastrSpy = jasmine.createSpyObj('ToastrService',
+      [
+        'success',
+        'error'
+      ]);
+  const iamSpy = jasmine.createSpyObj('iam', [
+    'getDid',
+    'checkExistenceOfDomain',
+    'isOwner',
+    'getRoleDIDs',
+    'createRole',
+    'setRoleDefinition',
+    'getENSTypesBySearchPhrase'
+  ]);
+  const iamServiceSpy = jasmine.createSpyObj('iam', [
+    'isAlphaNumericOnly',
```

Comment:
This method in iamservice is deprecated. Please use function from `utils/functions/is-alpha-numeric`. That way we could remove this spy. 
And i know it is the same code, but i started removing parts in IamService that have nothing to do with IAM object from iam-client-lib.